File: app/src/main/java/com/mnj/java_solutions/linkedlist/ReverseLinkedList.java
package com.mnj.java_solutions.linkedlist;

public class ReverseLinkedList {

    public void reverseLinkedList() {

        ListNode node = new ListNode();
        ListNode.Node head = node.insert();

        ListNode.Node curr = head;
        ListNode.Node prev = null;

        while (curr != null) {
            ListNode.Node next = curr.next;
            curr.next = prev;
            prev = curr;
            curr = next;
        }
        node.display(prev);
    }
}

File: app/src/main/java/com/mnj/java_solutions/linkedlist/ReverseLinkedList.kt
package com.mnj.java_solutions.linkedlist

import com.mnj.java_solutions.linkedlist.ListNode

class ReverseLinkedList {
    fun reverseLinkedList() {
        val node = ListNode()
        val head = node.insert()
        var curr = head
        var prev: ListNode.Node? = null
        while (curr != null) {
            val next = curr.next
            curr.next = prev
            prev = curr
            curr = next
        }
        node.display(prev)
    }
}